### PR TITLE
rm old discrete system

### PR DESCRIPTION
**Pull request by baggepinnen**

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
Rebase

File: src/utils.jl [hunk 1]
```diff
@@ -319,23 +315,6 @@ isoperator(op) = expr -> isoperator(expr, op)
 isdifferential(expr) = isoperator(expr, Differential)
 isdiffeq(eq) = isdifferential(eq.lhs)
 
-isdifference(expr) = isoperator(expr, Difference)
-isdifferenceeq(eq) = isdifference(eq.lhs)
```
Comment (AayushSabharwal) [inline src/utils.jl:323]: These functions are used in other places. Is there a corresponding implementation for the new discrete system? `hasshift`?

Comment (baggepinnen) [reply src/utils.jl:323]: `isdifference` is no longer the correct question to ask, you can have an equation that is not a difference equation, but it still belongs to the discrete time domain. 

You now have `has_discrete_domain` and other similar functions in the same file, but they require clock inference to have been run on the model before they are fully usable.

**Comment by AayushSabharwal** [comment]
I don't have the permissions to push here, so I opened another PR branched off of this one https://github.com/SciML/ModelingToolkit.jl/pull/2440

**Pull request closed without merging**